server/src/main/java/org/opensearch/indices/NodeIndicesStats.java

```diff
@@ -112,6 +119,83 @@ public NodeIndicesStats(CommonStats oldStats, Map<Index, List<IndexShardStats>>
         }
     }
 
+    /**
+     * Passing the level information to the nodes allows us to aggregate the stats based on the level passed. This
+     * allows us to aggregate based on NodeLevel (default - if no level is passed) or Index level if `indices` level is
+     * passed and finally return the statsByShards map if `shards` level is passed. This allows us to reduce ser/de of
+     * stats and return only the information that is required while returning to the client.
+     */
+    public NodeIndicesStats(
+        CommonStats oldStats,
+        Map<Index, List<IndexShardStats>> statsByShard,
+        SearchRequestStats searchRequestStats,
+        String[] levels
```

Comment:
Can we move the level array validation to caller and accept only one level (`String level`) here ?

Reply:
We can add a validation to the caller to accept only 1 level but that is anyways done on the toXContent method - 
See - https://github.com/opensearch-project/OpenSearch/pull/14454/files#diff-b79bb7aee566fa9c63b9f972aaa2b26206e0717a7e7d831307c0794161ec64bfR326-R337